pkg/types/azure/platform.go

```diff
@@ -0,0 +1,30 @@
+package azure
+
+import "strings"
+
+// Platform stores all the global configuration that all machinesets
+// use.
+type Platform struct {
+	// Region specifies the Azure region where the cluster will be created.
+	Region string `json:"region"`
+
+	// BaseDomainResourceGroupName specifies the resource group where the azure DNS zone for the base domain is found
```

Comment:
https://docs.microsoft.com/en-us/rest/api/dns/zones/list

It looks like we can query for DNS zones... why would we want to ask user to specify that?

Reply:
it's not about asking the user, I AM querying for it, but I need to store it somewhere for the TFVars to use it. if not here, where? I asked that in the other comment already